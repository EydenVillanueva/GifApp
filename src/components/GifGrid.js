import React from 'react';
import PropTypes from 'prop-types';

import GifGridItem from "./GifGridItem";
import {useFetchGifs} from "../hooks/useFetchGifs";

const GifGrid = ({category}) => {
	
	const { loading, data: images } = useFetchGifs(category);
	
	const renderImages = (images) => {
		return <ol>
			{ images.map( ({id, title, url}) => <GifGridItem key={id} title={title} url={url} /> )}
		</ol>;
	}
	
	return (
		<div className="card-grid" >
			<h3> {category} </h3>
			{ loading && <p>'Loading...'</p> }
			{renderImages(images)}
		</div>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired
};

export default GifGrid;


