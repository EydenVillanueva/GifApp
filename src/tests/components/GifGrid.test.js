import React from 'react';
import {shallow} from "enzyme";

import GifGrid from "../../components/GifGrid";
import {useFetchGifs} from "../../hooks/useFetchGifs";
jest.mock('../../hooks/useFetchGifs');


describe('GifGrid component test suite', () => {
	let category = 'category';
	let wrapper;
	
	
	test('GifGrid component should match snapshot', () => {
		useFetchGifs.mockReturnValue({
			data: [],
			loading: true,
		});
		
		wrapper = shallow( <GifGrid category={category} /> );
		expect(wrapper).toMatchSnapshot();
	});
	
	test('GifGrid should show items when image are loaded with useFetchGifs', () => {
		
		const gifs = [
			{ id: 1, title: 'Image title', url: 'http://localhost.com/image1.jpg' },
			{ id: 2, title: 'Image title 2', url: 'http://localhost.com/image2.jpg' },
		]
		
		useFetchGifs.mockReturnValue({
			data: gifs,
			loading: false,
		});
		
		wrapper = shallow( <GifGrid category={category} /> );
		
		expect( wrapper ).toMatchSnapshot();
		expect( wrapper.find('p').exists() ).toBe(false);
		expect( wrapper.find('GifGridItem').length ).toBe(gifs.length);
	});
});