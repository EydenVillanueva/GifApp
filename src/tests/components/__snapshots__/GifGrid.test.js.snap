// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid component test suite GifGrid component should match snapshot 1`] = `
<div
  className="card-grid"
>
  <h3>
     
    category
     
  </h3>
  <p>
    'Loading...'
  </p>
  <ol />
</div>
`;

exports[`GifGrid component test suite GifGrid should show items when image are loaded with useFetchGifs 1`] = `
<div
  className="card-grid"
>
  <h3>
     
    category
     
  </h3>
  <ol>
    <GifGridItem
      key="1"
      title="Image title"
      url="http://localhost.com/image1.jpg"
    />
    <GifGridItem
      key="2"
      title="Image title 2"
      url="http://localhost.com/image2.jpg"
    />
  </ol>
</div>
`;
